### PR TITLE
Nengo master compatibility

### PR DESCRIPTION
**Pull request by hunse**
Get working with the `master` branch of Nengo. 

**TODO**

- [x] Test

### CODE REVIEW

**Comment by tbekolay** [comment]
Task linked: [CU-bc9g2v nengo-ocl: Minor features/fixes in examples](https://app.clickup.com/t/bc9g2v)

File: nengo_ocl/builder.py [hunk 1]
```diff
@@ -115,3 +117,38 @@ def build_pes(model, pes, rule):
...
+            pass
+
+        return step
+
+
+@Builder.register(Probe)
+def build_probe(model, probe):
+    """Overridden build_probe to add ProbeOperator to track probed signals"""
+    nengo.builder.probe.build_probe(model, probe)
+    model.add_op(ProbeOperator(model.sig[probe]["in"]))
```
Comment (drasmuss) [inline nengo_ocl/builder.py:89]: This is a nice solution, cleaner than what I'm doing in NengoDL

**Review by drasmuss — APPROVED** [review]
Looks good

**Comment by hunse** [comment]
This is now fixed to work with Nengo v3.1.0. I've modified it to use the new `SimProbe` operator, raise a NotImplementedError for the RLS learning rule, removed the `Signal.may_share_memory` patch (since that's now fixed in Nengo), and applied `isort`.

Ready for merge at your leisure, @drasmuss.

**Pull request merged**